Uninitialized variable warnings

**Describe the bug**

It would be nice to fix these:

```
src/libstore/content-address.cc: In static member function ‘static nix::ContentAddressMethod nix::ContentAddressMethod::parsePrefix(std::string_view&)’:
src/libstore/content-address.cc:36:16: warning: ‘method’ may be used uninitialized [-Wmaybe-uninitialized]
   36 |         method = TextIngestionMethod {};
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
src/libstore/content-address.cc:32:26: note: ‘method’ declared here
   32 |     ContentAddressMethod method = FileIngestionMethod::Flat;
      |                          ^~~~~~
```

and

```
src/libstore/derivations.cc: In function ‘nix::Derivation nix::parseDerivation(const Store&, std::string&&, std::string_view, const ExperimentalFeatureSettings&)’:
src/libstore/derivations.cc:384:98: warning: ‘version’ may be used uninitialized [-Wmaybe-uninitialized]
  384 |         drv.inputDrvs.map.insert_or_assign(store.parseStorePath(drvPath), parseDerivedPathMapNode(store, str, version));
      |                                                                           ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
src/libstore/derivations.cc:350:28: note: ‘version’ was declared here
  350 |     DerivationATermVersion version;
      |                            ^~~~~~~
```

--- a/src/libcmd/command.cc
+++ b/src/libcmd/command.cc
@@ -175,7 +175,7 @@ void BuiltPathsCommand::run(ref<Store> store, Installables && installables)
             throw UsageError("'--all' does not expect arguments");
         // XXX: Only uses opaque paths, ignores all the realisations
         for (auto & p : store->queryAllValidPaths())
-            paths.push_back(BuiltPath::Opaque{p});
+            paths.emplace_back(BuiltPath::Opaque{p});
     } else {
         paths = Installable::toBuiltPaths(getEvalStore(), store, realiseMode, operateOn, installables);
         if (recursive) {
@@ -188,7 +188,7 @@ void BuiltPathsCommand::run(ref<Store> store, Installables && installables)
             }
             store->computeFSClosure(pathsRoots, pathsClosure);
             for (auto & path : pathsClosure)
-                paths.push_back(BuiltPath::Opaque{path});
+                paths.emplace_back(BuiltPath::Opaque{path});
         }
     }
 

--- a/src/libcmd/installables.cc
+++ b/src/libcmd/installables.cc
@@ -663,7 +663,7 @@ BuiltPaths Installable::toBuiltPaths(
 
         BuiltPaths res;
         for (auto & drvPath : Installable::toDerivations(store, installables, true))
-            res.push_back(BuiltPath::Opaque{drvPath});
+            res.emplace_back(BuiltPath::Opaque{drvPath});
         return res;
     }
 }

--- a/src/libstore/derivations.cc
+++ b/src/libstore/derivations.cc
@@ -352,7 +352,7 @@ Derivation parseDerivation(
         expect(str, "erive(");
         version = DerivationATermVersion::Traditional;
         break;
-    case 'r':
+    case 'r': {
         expect(str, "rvWithVersion(");
         auto versionS = parseString(str);
         if (versionS == "xp-dyn-drv") {
@@ -364,6 +364,9 @@ Derivation parseDerivation(
         }
         expect(str, ",");
         break;
+    }
+    default:
+        throw Error("derivation does not start with 'Derive' or 'DrvWithVersion'");
     }
 
     /* Parse the list of outputs. */

--- a/src/libstore/store-api.cc
+++ b/src/libstore/store-api.cc
@@ -819,7 +819,7 @@ void Store::substitutePaths(const StorePathSet & paths)
     std::vector<DerivedPath> paths2;
     for (auto & path : paths)
         if (!path.isDerivation())
-            paths2.push_back(DerivedPath::Opaque{path});
+            paths2.emplace_back(DerivedPath::Opaque{path});
     uint64_t downloadSize, narSize;
     StorePathSet willBuild, willSubstitute, unknown;
     queryMissing(paths2,

--- a/src/nix-build/nix-build.cc
+++ b/src/nix-build/nix-build.cc
@@ -449,7 +449,7 @@ static void main_nix_build(int argc, char * * argv)
             }
         }
         for (const auto & src : drv.inputSrcs) {
-            pathsToBuild.push_back(DerivedPath::Opaque{src});
+            pathsToBuild.emplace_back(DerivedPath::Opaque{src});
             pathsToCopy.insert(src);
         }
 

--- a/src/nix-env/nix-env.cc
+++ b/src/nix-env/nix-env.cc
@@ -481,12 +481,12 @@ static void printMissing(EvalState & state, DrvInfos & elems)
     std::vector<DerivedPath> targets;
     for (auto & i : elems)
         if (auto drvPath = i.queryDrvPath())
-            targets.push_back(DerivedPath::Built{
+            targets.emplace_back(DerivedPath::Built{
                 .drvPath = makeConstantStorePathRef(*drvPath),
                 .outputs = OutputsSpec::All { },
             });
         else
-            targets.push_back(DerivedPath::Opaque{
+            targets.emplace_back(DerivedPath::Opaque{
                 .path = i.queryOutPath(),
             });
 

